tools/dotnet-linker/Steps/ManagedRegistrarStep.cs

```diff
@@ -921,14 +921,42 @@ bool EmitConversion (MethodDefinition method, ILProcessor il, TypeReference type
 						// cast to the generic type to verify that the item is actually of the correct type
 						il.Emit (OpCodes.Unbox_Any, type);
 					} else {
-						var nativeObjType = StaticRegistrar.GetInstantiableType (type.Resolve (), exceptions, GetMethodSignature (method));
+						StaticRegistrar.GetInstantiableType (type.Resolve (), exceptions, GetMethodSignature (method), out var ctor);
+						EnsureVisible (method, ctor);
+						var targetType = method.Module.ImportReference (type);
+						var handleVariable = il.Body.AddVariable (abr.System_IntPtr);
+						var objectVariable = il.Body.AddVariable (targetType);
+						var loadHandle = il.Create (OpCodes.Ldloc, handleVariable);
+						var loadObjectVariable = il.Create (OpCodes.Ldloc, objectVariable);
+						il.Emit (OpCodes.Stloc, handleVariable);
+						// objectVariable = null
+						il.Emit (OpCodes.Ldnull);
+						il.Emit (OpCodes.Stloc, objectVariable);
+						// if (handle == IntPtr.Zero)
+						//     goto done;
+						il.Emit (OpCodes.Ldloc, handleVariable); // handle
+						il.Emit (OpCodes.Ldsfld, abr.System_IntPtr_Zero);
+						il.Emit (OpCodes.Beq, loadObjectVariable);
+						// objectVariable = TryGetNSObject (handle, false) as TargetType
+						il.Emit (OpCodes.Ldloc, handleVariable); // handle
+						il.Emit (OpCodes.Ldc_I4_0); // false
+						il.Emit (OpCodes.Call, abr.Runtime_TryGetNSObject);
+						il.Emit (OpCodes.Castclass, targetType);
+						il.Emit (OpCodes.Stloc, objectVariable);
```

Comment:
if you insert `il.Emit (OpCodes.Dup)` before the stloc, you can eliminate the ldloc. It probably will AOT into the same code, but still.

Reply:
I was doing it that way at one point, but then changed since I figured the current way is more readable and easier to reason about.